library_generation/utilities.sh

```diff
@@ -273,3 +273,19 @@ get_version_from_versions_txt() {
   version=$(grep "$key:" "${versions}" | cut -d: -f3) # 3rd field is snapshot
   echo "${version}"
 }
+
+detect_OS() {
+  if [[ "${OSTYPE}" == "linux-gnu"* ]] || [[ "${OSTYPE}" == "freebsd"* ]]; then
+    os_architecture="linux-x86_64"
+  elif [[ "${OSTYPE}" == "darwin"* ]]; then
```

Comment:
What about `osx-aarm_64`?

I think you can use `uname -sm` to detect OS type and CPU architecture, but I'm not sure whether windows can run this command.

Reply:
I defaulted to `win32` regardless of architecture and changed it to `uname -m` to detect the architecture only, while keep reliance on `${OSTYPE}`